src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
But if you subsample different points each time you either converge to the optimum split plane, which means your different random trees wouldn't be so random anymore, or you don't converge at all... I think the benefit from the random forest comes from sampling a specific number of points, and as Chris says, you might as well just keep a small sample, and maybe that sample being 2 points is good enough anyway.